include/onnxruntime/core/framework/data_types.h

```diff
@@ -508,10 +562,84 @@ class SparseTensorType : public SparseTensorTypeBase {
  private:
   SparseTensorType() {
     using namespace data_types_internal;
-    TensorElementTypeSetter<elemT>::SetSparseTensorElementType(mutable_type_proto());
+    TensorElementTypeSetter<elemT>::SetSparseTensorElementType(MutableTypeProto());
+  }
+};
+
+/// Common base-class for all optional types.
+class OptionalTypeBase : public DataTypeImpl {
+ public:
```

Comment:
What is the binary size impact of adding support for Optional types? Should it be able to be disabled the same as sparse tensor support?

Reply:
Had already planned to support disabling. But, I will do it in a separate change after this is merged.